src/vcpkg/postbuildlint.cpp

```diff
@@ -534,11 +534,16 @@ namespace vcpkg::PostBuildLint
 
 #if defined(_WIN32)
 
-    static LintStatus check_dll_architecture(const std::string& expected_architecture,
+    static LintStatus check_dll_architecture(const std::vector<std::string>& expected_architectures,
```

Comment:
I think we should check that all the architectures are present now, what do you think?

Reply:
I assumed when writing this code that windows dlls only supported a single architecture, since that was the existing behavior. I therefore added the assert below to indicate that the method only supports a single architecture, and if we start compiling "unified" dlls on windows we'll need another look at the code.

The ARM64EC machine type looks similar to Macs unified binaries, but it doesn't look like they solved it the same way?